pkg/policy/l4.go

```diff
@@ -43,6 +50,89 @@ type L4Filter struct {
 	Ingress bool
 }
 
+// FillPortRuleHTTP fills in the http port rule fields
+// as a regex into a single string
+func FillPortRuleHTTP(httpRule []api.PortRuleHTTP) []AuxRule {
+	l7rules := []AuxRule{}
+	for _, h := range httpRule {
+		r := AuxRule{L7Parser: HTTPStr}
+
+		if h.Path != "" {
+			r.Expr = "PathRegexp(\"" + h.Path + "\")"
+		}
+
+		if h.Method != "" {
+			if r.Expr != "" {
+				r.Expr += " && "
+			}
+			r.Expr += "MethodRegexp(\"" + h.Method + "\")"
+		}
+
+		if h.Host != "" {
+			if r.Expr != "" {
+				r.Expr += " && "
+			}
+			r.Expr += "HostRegexp(\"" + h.Host + "\")"
+		}
+
+		for _, hdr := range h.Headers {
+			s := strings.SplitN(hdr, " ", 2)
+			if r.Expr != "" {
+				r.Expr += " && "
+			}
+			r.Expr += "Header(\""
+			if len(s) == 2 {
+				// Remove ':' in "X-Key: true"
+				key := strings.TrimRight(s[0], ":")
+				r.Expr += key + "\",\"" + s[1]
+			} else {
+				r.Expr += s[0]
+			}
+			r.Expr += "\")"
+		}
+
+		if r.Expr != "" {
+			l7rules = append(l7rules, r)
+		}
+	}
+
+	return l7rules
+}
+
+// FillPortRuleKafka fills in the Kafka port rule fields
+// as a regex into a single string
+// TODO The string formed is likely not the final format, and will
+// evolve as more kafka keys are supported.
+func FillPortRuleKafka(kafkaRule []api.PortRuleKafka) []AuxRule {
```

Comment:
I think whatever parser and enforcer we have, can directly understand `api.PortRuleKafka` so we can probably skip this step and simply pass in the validated rules.

Reply:
If you mean the Vulcand proxy router? then not really, it is specific to http and all the router checks assume http requests only  and does not understand Kafka at all. I think we need to write the router from scratch for Kafka, as well as the parser, which is what I am working on.


I am trying to see if I can reuse some Vulcand proxy router functionality, and will leave this one as is for now.

I can change it later if I feel its not really needed.